src/ReaderThemeSupportFeatures.php

```diff
@@ -219,9 +294,29 @@ public function get_theme_support_features( $reduced = false ) {
 				$features[ $feature_key ] = $feature_value;
 			}
 		}
+
 		return $features;
 	}
 
+	/**
+	 * Check if theme.json can be used to determine theme support.
+	 * Due to the lack of `wp_get_global_settings()`, this will always return false for WP < 5.9.
+	 *
+	 * @since 2.4.1
+	 *
+	 * @return bool Whether theme.json can be used to determine theme support.
+	 */
+	private function maybe_use_theme_json() {
+		// wp_get_global_settings() is only available in WP 5.9+.
+		if ( ! function_exists( 'wp_get_global_settings' ) ) {
+			return false;
+		}
+
+		// Do not rely on `wp_is_block_theme()` as theme.json can be used in non-block themes.
+		// See: https://developer.wordpress.org/themes/advanced-topics/theme-json/#a-theme-json-can-be-added-to-any-theme.
+		return ( is_readable( get_stylesheet_directory() . '/theme.json' ) ? true : is_readable( get_template_directory() . '/theme.json' ) );
```

Comment:
This would seem simpler as:

```suggestion
		return is_readable( get_stylesheet_directory() . '/theme.json' ) || is_readable( get_template_directory() . '/theme.json' );
```

Reply:
Actually, I suggest using `wp_theme_has_theme_json()` if it exists, or else copy the function as a method into this class:

https://github.com/WordPress/wordpress-develop/blob/200868214a1ae0a108dac491677ba82e7541fc8d/src/wp-includes/global-styles-and-settings.php#L384-L414

Note how it caches the file exists check.

Reply:
> Note how it caches the file exists check.

Oh yes. Bette to copy it since we will need to check for theme.json on WP 5.9+ and this function is added in WP 6.2.